conda/core/solve.py

```diff
@@ -653,22 +639,21 @@ def _add_specs(self, ssc):
                                       for prec in ssc.prefix_data.iter_records()
                                       )
 
+        # ensure that our self.specs_to_add are not being held back by packages in the env.
+        #    This factors in pins and also ignores specs from the history.  It is unfreezing only
+        #    for the indirect specs that otherwise conflict with update of the immediate request
         elif ssc.update_modifier == UpdateModifier.UPDATE_SPECS:
-            for spec in self.specs_to_add:
-                spec = MatchSpec(spec)
-                if (spec.name in pin_overrides and not ssc.ignore_pinned
-                        or spec.name in ssc.specs_from_history_map):
-                    # skip this spec, because it is constrained by pins
-                    continue
-                ssc.specs_map[spec.name] = spec
-                # the index is sorted, so the first record here gives us what we want.
-                latest_pkg = ssc.r.find_matches(spec)[0]
-                for ms in list(ssc.specs_map.values()):
-                    ms = MatchSpec(ms)
-                    spec_pool = ssc.r._get_package_pool([ms])
-                    if spec_pool.get(spec.name) and latest_pkg not in spec_pool[spec.name]:
-                        # neuter the spec due to a conflict
-                        ssc.specs_map[ms.name] = MatchSpec(ms.name)
+            skip = lambda x: (x.name in pin_overrides and not ssc.ignore_pinned
+                              or x.name in ssc.specs_from_history_map)
+            specs_to_add = [self._package_has_updates(ssc, _, installed_pool)
+                            for _ in self.specs_to_add if not skip(_)]
+            # the index is sorted, so the first record here gives us what we want.
+            conflicts = ssc.r.get_conflicting_specs(specs_to_add +
```

Comment:
condensing this to one call to get_conflicting_specs is a speedup relative to the loop with comparing pools.  That one call is slower than any given pool comparison, but faster overall.